components/restapi/core/accounts.go

```diff
@@ -0,0 +1,203 @@
+package core
+
+import (
+	"time"
+
+	"github.com/labstack/echo/v4"
+
+	"github.com/iotaledger/hive.go/ierrors"
+	"github.com/iotaledger/inx-app/pkg/httpserver"
+	"github.com/iotaledger/iota-core/pkg/core/account"
+	restapipkg "github.com/iotaledger/iota-core/pkg/restapi"
+	iotago "github.com/iotaledger/iota.go/v4"
+	"github.com/iotaledger/iota.go/v4/nodeclient/apimodels"
+)
+
+func blockIssuanceCreditsForAccountID(c echo.Context) (*apimodels.BlockIssuanceCreditsResponse, error) {
+	accountID, err := httpserver.ParseAccountIDParam(c, restapipkg.ParameterAccountID)
+	if err != nil {
+		return nil, err
+	}
+	slotIndex, err := httpserver.ParseSlotQueryParam(c, restapipkg.ParameterSlotIndex)
+	if err != nil {
+		// by default we return the balance for the latest slot
+		slotIndex = deps.Protocol.SyncManager.LatestCommittedSlot()
+	}
+	account, exists, err := deps.Protocol.MainEngineInstance().Ledger.Account(accountID, slotIndex)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, ierrors.Errorf("account not found: %s", accountID.ToHex())
+	}
+
+	return &apimodels.BlockIssuanceCreditsResponse{
+		SlotIndex:            slotIndex,
+		BlockIssuanceCredits: account.Credits.Value,
+	}, nil
+}
+
+func congestionForAccountID(c echo.Context) (*apimodels.CongestionResponse, error) {
+	accountID, err := httpserver.ParseAccountIDParam(c, restapipkg.ParameterAccountID)
+	if err != nil {
+		return nil, err
+	}
+	mca := deps.Protocol.CurrentAPI().ProtocolParameters().EvictionAge()
+	slotIndex := deps.Protocol.CurrentAPI().TimeProvider().SlotFromTime(time.Now())
+	if slotIndex < mca {
+		mca = 0
+	}
+	account, exists, err := deps.Protocol.MainEngineInstance().Ledger.Account(accountID, slotIndex-mca)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, ierrors.Errorf("account not found: %s", accountID.ToHex())
+	}
+
+	return &apimodels.CongestionResponse{
+		SlotIndex:            slotIndex,
+		Ready:                false, // TODO: update after scheduler is implemented
+		ReferenceManaCost:    0,     // TODO: update after RMC is implemented
+		BlockIssuanceCredits: account.Credits.Value,
+	}, nil
+}
+
+func staking() (*apimodels.AccountStakingListResponse, error) {
```

Comment:
I thought we need an endpoint that returns all possible stakes, instead of the active ones? Or is this exactly that?
Does activeValidators here mean that they were active in the last activity window?